### PR TITLE
Basic native push

### PR DESCRIPTION
**Pull request by jpatel531**
cc @vivangkumar @zimbatm 

TODO:

- [x] tests
- [x] validations

### CODE REVIEW

**Comment by zimbatm** [comment]
Did you have a look at vivan's implementation in the ruby lib?

**Comment by jpatel531** [comment]
Yep. Thats my guideline. Todos yet to be done though

**Comment by zimbatm** [comment]
Does it mean you'll make a separate class and the hostname configurable too?

**Comment by jpatel531** [comment]
Yeah I will

File: pusher/pusher.py [hunk 1]
```diff
@@ -13,11 +15,12 @@
 import re
 import six
 import time
+import copy
```
Comment (zimbatm) [inline pusher/pusher.py]: I don't think that's used anymore?

**Pull request merged**